src/kernels/raw/finder/localKernelSpecFinderBase.node.ts

```diff
@@ -188,31 +193,48 @@ export abstract class LocalKernelSpecFinderBase {
         paths: (Uri | { interpreter: PythonEnvironment; kernelSearchPath: Uri })[],
         cancelToken?: CancellationToken
     ): Promise<KernelSpecFileWithContainingInterpreter[]> {
-        const searchResults = await Promise.all(
-            paths.map(async (searchItem) => {
-                const searchPath = isUri(searchItem) ? searchItem : searchItem.kernelSearchPath;
-                if (await this.fs.localDirectoryExists(searchPath.fsPath)) {
-                    const files = await this.fs.searchLocal(`**/kernel.json`, searchPath.fsPath, true);
```

Comment:
If you look at the logs on CI, we end up searching for kernels every single time in each interpreter.
& the number of interpreters seems to grow by 1 for each test.
This ensures we cache the promise for each `path` argument, & even if we have duplicates it will be handled as we generate unique keys and cache it by each `path` argument.